### ISSUE TITLE
tester().waitForView(withAccessibilityLabel: text, traits: UIAccessibilityTraitButton). Issues 

### ISSUE THREAD

**Issue by readwrite123455** [kif-framework/KIF#1091]
tester().waitForView(withAccessibilityLabel: text, traits: UIAccessibilityTraitButton)
    tester().tapView(withAccessibilityLabel: text, traits:UIAccessibilityTraitButton) . 

 I am getting           caught "NSInvalidArgumentException", "*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil" 
Any one can help me out. Already working from last 7 days .

Any help would be appraciated.

**Comment by justinseanmartin** [comment]
This is not enough information to give guidance. What is the full stack trace when the assertion happens?

FYI - You shouldn't need to perform a `waitForView` before `tapView`. KIF will spin for up to 10 seconds trying to find a matching element on the screen before taking action on it.

**Comment by readwrite123455** [comment]
The step timed out after 10.00 seconds: Failed to find accessibility element with the label "User Login Successfully" In  tester().waitForView(withAccessibilityLabel: text) ExpectToseealert method. How to show alert pop up .

**Comment by readwrite123455** [comment]
Any help would be appreciated

**Comment by readwrite123455** [comment]
The step timed out after 60.00 seconds: Failed to find accessibility element with the label "Please Enter Valid Email"

**Comment by mikelupo** [comment]
@readwrite123455 because you're not supplying enough information to help us help you, I will just tell you what I suspect. 1) the popup isn't happening. 2) the popup is happening, but it's not owned by the your app, instead it's owned by some other app or system (System meaning Apple) and you're up against Apple security, 3) You have a mismatch in spelling or punctuation in the accessibility label/identifier, 4) The author set the accessibility identifier, but you're checking for accessibility label (or visa-versa).

I've been using KIF for a number of years and these sorts of issues are most always one of those items I mentioned above.
I hope this helps. If you want any further help, I suggest you write a small UI app that exhibits the problem, share it on github a then provide a link here. We're happy to help.

**Comment by justinseanmartin** [comment]
This sounds like a different issue than the original issue you posted. What changed?

I'd recommend setting a breakpoint in the code where you expect the alert to be getting presented and try to figure out if/why that code isn't being executed.

Some other debugging tips:
* In the LLDB debug console, you can try typing `po [UIView printViewHierarchy]` to dump the current state of the accessibility hierarchy.
* You can also turn on the "Accessibility Inspector" tool from the Xcode window to see accessibility of elements.
* Your reply sounds like a different issue, but when I asked about a stack trace, you could type "bt" into the debug window and copy that.

Agree with @mikelupo, please create a repo with a sample app exhibiting the behavior so we can see what's going on, rather than guessing.

**Comment by readwrite123455** [comment]
Hi Justin and Mike ,

You are correct previous issues was my accessibility object are not accessible .
 I resolved that issues by giving Accessibility traits to button. 
Now I want to show error in popup (AlertView ).
 I am using expectToSeeAlert method of KIF . But Pop up is not displaying. 
Can you please help me to resolve this issues? 
Any help would be appreciated. 
 Or Is there any alternative So that I can show  UIAlertView.

**Comment by mikelupo** [comment]
@readwrite123455 It sounds to me that the pop up not displaying is an issue with the app you’re testing and not your test or KIF. 
It might be better served to post that question on stackoverflow.com.
I’m going to close this but feel free to reopen if you’re certain that KIF is the problem. We’re here to help.

**Comment by readwrite123455** [comment]
ok It means pop up will use in App itself not from KIF .
Please correct me If I am worong. Thanks for quick reply.

**Comment by justinseanmartin** [comment]
I'm not sure what you mean. Can you please clarify or restate the question? 

If the alert is one that you're presenting within your app and not a system alert for permissions (e.g. location, camera, microphone, etc), then you should be able to interact with it from KIF just fine.
